torch/csrc/jit/interned_strings.h

```diff
@@ -3,146 +3,252 @@
-_(zeros) \
-_(__ATEN_EXTRA_END)
+_(aten, __not__) \
+/* end */
+
+#define FORALL_ATEN_SYMBOLS(_) \
+FORALL_ATEN_BASE_SYMBOLS(_) \
+FORALL_ATEN_EXTRA_SYMBOLS(_)
 
 // These symbols correspond to ONNX operators.  Their semantics
 // are defined in https://github.com/onnx/onnx/blob/master/docs/Operators.md
 // The particular version we are targeting is specified by '_onnx_opset_version'
 // in torch.onnx.symbolic
+//
+// In general, most ONNX operators won't get an entry here, because they
+// are handled from the Python end.  However, you may occasionally need
+// to intern an ONNX symbol here so that you can conveniently write an
+// optimization on ONNX operations.
 
 #define FORALL_ONNX_SYMBOLS(_) \
-_(Add) \
-/* _(Constant) conflicts with JIT */ \
-_(Div) \
-_(GRU) \
-_(Gemm) \
-_(LSTM) \
-_(Mul) \
-_(PackPadded) \
-_(PadPacked) \
-_(Pow) \
-_(RNN) \
-_(Slice) /* used by test only */ \
-_(Sub) \
-_(Transpose) \
-_(__ONNX_END)
+_(onnx, Add) \
+_(onnx, Constant) \
+_(onnx, Div) \
+_(onnx, GRU) \
+_(onnx, Gemm) \
+_(onnx, LSTM) \
+_(onnx, Mul) \
+_(onnx, Pow) \
+_(onnx, RNN) \
+_(onnx, Sub) \
+_(onnx, Transpose) \
+/* end */
 
 // These symbols are attribute keys.  They are shared between both ONNX and ATen
-// operators (you disambiguate their meaning by looking at the operator itself)
+// operators (you disambiguate their meaning by looking at the operator itself).
+// In general, you only need to define attribute keys that are used by
+// onnx or prim; ATen attributes are automatically generated in FORALL_ATTR_BASE_SYMBOLS.
+
+#define FORALL_ATTR_EXTRA_SYMBOLS(_) \
+_(attr, Subgraph) \
+_(attr, axis) \
+_(attr, broadcast) \
+_(attr, device) \
+_(attr, inplace) \
+_(attr, is_zero) \
+_(attr, perm) \
+_(attr, sizes) \
+_(attr, transA) \
+_(attr, transB) \
+/* end */
 
 #define FORALL_ATTR_SYMBOLS(_) \
-_(Subgraph) \
-_(alpha) \
-_(axis) \
-_(broadcast) \
-_(device) \
-/* _(dim) conflicts with ATen */ \
-_(end) \
-_(exponent) \
-_(inplace) \
-_(is_zero) \
-_(keepdim) \
-_(length) \
-_(other) \
-_(perm) \
-/* _(size) conflicts with ATen */ \
-/* _(sizes) conflicts with ATen */ \
-_(start) \
-_(step) \
-_(transA) \
-_(transB) \
-_(value) \
-_(__ATTR_END)
+FORALL_ATTR_BASE_SYMBOLS(_) \
+FORALL_ATTR_EXTRA_SYMBOLS(_)
 
 #define FORALL_BUILTIN_SYMBOLS(_) \
-FORALL_JIT_SYMBOLS(_) \
-FORALL_ATEN_SYMBOLS(_) \
-FORALL_ATEN_EXTRA_SYMBOLS(_) \
-FORALL_ONNX_SYMBOLS(_) \
-FORALL_ATTR_SYMBOLS(_) \
-_(__BUILTIN_END)
-
-  enum BuiltinSymbol {
-    #define DEFINE_SYMBOL(s) \
-      k##s,
-    FORALL_BUILTIN_SYMBOLS(DEFINE_SYMBOL)
-    #undef DEFINE_SYMBOL
-    kLastSymbol, //where we start counting for new symbols
-  };
+  FORALL_ONNX_SYMBOLS(_) \
+  FORALL_ATEN_SYMBOLS(_) \
+  FORALL_ATTR_SYMBOLS(_) \
+  FORALL_PRIM_SYMBOLS(_) \
+  /* end */
+
+// Note [Symbol allocation]
+// ~~~~~~~~~~~~~~~~~~~~~~~~
+//
+//  1. Symbol namespace is split up into namespaces.  The byte structure
+//  of our symbols is TUUUUUUU, where T is the tag byte and U are the unique
```

Comment:
We're using `uint32_t`, so you only have 4 bytes

Reply:
Oops, I can't math. That means we only have 16777216 uniques.

Reply:
Oh no, idk if that's enough 🤔

Reply:
You jest, but it's likely enough that I should probably add an overflow test XD